### DISCUSSION TITLE
Recommended pattern for fetching only if data is not cached?

### DISCUSSION THREAD

**Discussion by mjhuff** [TanStack/query#8250] [Q&A] [answered]
Hello,

What is the recommended pattern for first checking if cached data exists for a query key, and if not, then performing the query fetch? 

Bulletproof React offers [this pattern](https://github.com/alan2207/bulletproof-react/blob/9010f0d5943fccb7cec51aecbd4c1ee33703f26e/apps/react-vite/src/app/routes/app/users.tsx#L8-L15), but [reading the docs](https://tanstack.com/query/latest/docs/framework/react/guides/important-defaults), it seems that passing query options with `refetchOnMount: false` and `staleTime: Infinity` might accomplish the same thing?

I suppose one could do some sort of `() => queryClient.getQueryData(['key'])` coupled with an `enabled: !queryClient.getQueryData(['key'])` condition, too. 

Is there really no better solution than one of the methods above? It feels like this problem is common, and I'm overlooking a simpler method.

Thank you!

**Comment by TkDodo** [comment, marked as answer]
Just setting `staleTime: Infinity` is enough.
